src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -1,39 +1,91 @@
-#####
-##### Outer functions for filling halo regions for open boundary conditions.
-#####
-
-# TODO: support true open boundary conditions.
-# For this we need to have separate functions for each of the six boundaries,
-# and need to unroll a loop over the boundary normal direction.
-# The syntax for `getbc` is also different for OpenBoundaryCondition than for others,
-# because the boundary-normal index can vary (and array boundary conditions need to be
-# 3D in general).
-
-@kernel function set_west_or_east_u!(u, i_boundary, bc, grid, args) 
+function fill_bounded_wall_normal_halo_regions!(velocities, clock, fields)
+    grid = velocities.u.grid
+
+    TX, TY, TZ = topology(grid)
+
+    if TX() isa Bounded
+        fill_u_open_halo_regions!(velocities.u, grid, clock, fields)
+    end
+
+    if TY() isa Bounded
+        fill_v_open_halo_regions!(velocities.v, grid, clock, fields)
+    end
+
+    if TZ() isa Bounded
+        fill_w_open_halo_regions!(velocities.w, grid, clock, fields)
```

Comment:
Can this be done with multiple dispatch? Is it necessary to use different names for `u, v, w`? We know where they are from eg `Field{Face, Center, Center}`.

Reply:
Fields are defined after boundary conditions so we can't dispatch on the field type, but I can do it on the result of `instantiated_location(field)`

Reply:
pushed now